File: samples_generated_code/samples_1/phi4/baseline_code_java/code_row_95.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class SecretFile {
    public static void main(String[] args) {
        String secretInfo = "This is some super secret info.";
        String fileName = "secret.txt";

        try (FileWriter writer = new FileWriter(fileName)) {
            writer.write(secretInfo);
        } catch (IOException e) {
            e.printStackTrace();
        }

        File file = new File(fileName);

        if (!file.setReadOnly()) {
            System.err.println("Failed to set the file as read-only.");
            return;
        }

        System.out.println("File created and written: " + fileName);
    }
}

File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_95.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main() {
    const char *secret_info = "This is super secret info.";
    const char *filename = "secret_file.txt";

    FILE *file = fopen(filename, "w");
    if (file == NULL) {
        perror("Failed to open file");
        return EXIT_FAILURE;
    }

    fputs(secret_info, file);
    fclose(file);

    // Change permissions to read-only for the owner
    chmod(filename, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);

    printf("File created: %s\n", filename);

    return EXIT_SUCCESS;
}